### PR TITLE
[CWS] Add splice event type

### PR DESCRIPTION
**Pull request by Gui774ume**
### What does this PR do?

This PR introduces the `splice` event type in the CWS agent. This event exports entry and exit flag of the `fd_out` pipe (if it is a pipe), along with the file context of the `fd_in` file descriptor (if it is a file). Approvers were added to the pipe entry & exit flags and on the file part of the event.

The following rules can now be written:

```
- id: dirty_pipe
    description: A process exploited the dirty_pipe vulnerability
    expression: >-
      (splice.pipe_exit_flag & PIPE_BUF_FLAG_CAN_MERGE) == PIPE_BUF_FLAG_CAN_MERGE

  - id: dirty_pipe_suid
    description: A process exploited the dirty_pipe vulnerability to overwrite a SUID or SGID binary
    expression: >-
      (splice.pipe_exit_flag & PIPE_BUF_FLAG_CAN_MERGE) == PIPE_BUF_FLAG_CAN_MERGE && (splice.file.mode & S_ISGID > 0 || splice.file.mode & S_ISUID)

  - id: dirty_pipe_executable_file
    description: A process exploited the dirty_pipe vulnerability to overwrite an executable file
    expression: >-
      (splice.pipe_exit_flag & PIPE_BUF_FLAG_CAN_MERGE) == PIPE_BUF_FLAG_CAN_MERGE && (splice.file.mode & S_IXGRP > 0 || splice.file.mode & S_IXOTH || splice.file.mode & S_IXUSR)

  - id: dirty_pipe_bin
    description: A process exploited the dirty_pipe vulnerability to overwrite a critical system binary file
    expression: >-
      (splice.pipe_exit_flag & PIPE_BUF_FLAG_CAN_MERGE) == PIPE_BUF_FLAG_CAN_MERGE && (splice.file.path in [ ~"/bin/*", ~"/sbin/*", ~"/usr/bin/*", ~"/usr/sbin/*", ~"/usr/local/bin/*", ~"/usr/local/sbin/*", ~"/boot/**" ]) 

  - id: dirty_pipe_root
    description: A process exploited the dirty_pipe vulnerability to overwrite a file owned by root
    expression: >-
      (splice.pipe_exit_flag & PIPE_BUF_FLAG_CAN_MERGE) == PIPE_BUF_FLAG_CAN_MERGE && (splice.file.uid == 0 || splice.file.gid == 0)
```

### Motivation

This event type can be used to detect the `Dirty Pipe` vulnerability.

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [x] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [x] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [x] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by lebauce — APPROVED** [review]

**Review by djmitche — APPROVED** [review]
(only core-related file is for the SECURITY_AGENT_POLICIES_VERSION)

File: docs/cloud-workload-security/backend.md [hunk 1]
```diff
@@ -1244,6 +1248,37 @@ CWS logs have the following JSON schema:
...
+    "additionalProperties": false,
+    "type": "object"
+}
+
+{{< /code-block >}}
+
+| Field | Description |
+| ----- | ----------- |
+| `pipe_entry_flag` | entry flag of the fd_out pipe passed to the splice syscall |
+| `pipe_exit_flag` | exit flag of the fd_out pipe passed to the splice syscall |
```
Comment (jtappa) [inline docs/cloud-workload-security/backend.md:1279]:
```suggestion
| `pipe_entry_flag` | Entry flag of the fd_out pipe passed to the splice syscall |
| `pipe_exit_flag` | Exit flag of the fd_out pipe passed to the splice syscall |
```

File: docs/cloud-workload-security/backend.md [hunk 2]
```diff
@@ -1244,6 +1248,37 @@ CWS logs have the following JSON schema:
...
+{{< code-block lang="json" collapsible="true" >}}
+{
+    "required": [
+        "pipe_entry_flag",
+        "pipe_exit_flag"
+    ],
+    "properties": {
+        "pipe_entry_flag": {
+            "type": "string",
+            "description": "entry flag of the fd_out pipe passed to the splice syscall"
```
Comment (jtappa) [inline docs/cloud-workload-security/backend.md:1263]:
```suggestion
            "description": "Entry flag of the fd_out pipe passed to the splice syscall"
```

File: docs/cloud-workload-security/backend.md [hunk 3]
```diff
@@ -1244,6 +1248,37 @@ CWS logs have the following JSON schema:
...
+        },
+        "pipe_exit_flag": {
+            "type": "string",
+            "description": "exit flag of the fd_out pipe passed to the splice syscall"
```
Comment (jtappa) [inline docs/cloud-workload-security/backend.md:1267]:
```suggestion
            "description": "Exit flag of the fd_out pipe passed to the splice syscall"
```

**Review by jtappa — APPROVED** [review]
Some small edit suggestions for docs, but otherwise approved by our team 👍🏻

File: docs/cloud-workload-security/backend.md (hunk 1 above)
Comment (Gui774ume) [reply docs/cloud-workload-security/backend.md]: 👍🏻

File: docs/cloud-workload-security/backend.md (hunk 2 above)
Comment (Gui774ume) [reply docs/cloud-workload-security/backend.md]: 👍🏻

File: docs/cloud-workload-security/backend.md (hunk 3 above)
Comment (Gui774ume) [reply docs/cloud-workload-security/backend.md]: 👍🏻

**Pull request merged**